backend/uchicago/src/coursesearch_parser.py

```diff
@@ -0,0 +1,41 @@
+import collections
+import re
+import scrapydo
+
+from .course import Course
+from .section import Section
+
+from .coursesearch_spider import CourseSearchSpider
+
+class CourseSearch(object):
+  init = False
+  def __init__(self, label, id):
+    if not CourseSearch.init:
+      scrapydo.setup()
+      CourseSearch.init = True
+    self.label = label
+    self.id = id
+    
+  @property
+  def courses(self):
+  
+    settings = {
+        'USER_AGENT': 'Mozilla/4.0 (compatible; MSIE 7.0; Windows NT 5.1)',
+        'DOWNLOADER_MIDDLEWARES': {
+            'src.coursesearch_spider.StateManagerMiddleware': 543,
+        }
+    }
+    
+    spider = CourseSearchSpider
+    ret = scrapydo.run_spider(spider, settings=settings, label = self.label, id = self.id)
```

Comment:
Ok, sounds good, this otherwise looks good to merge, so if you want to add the sharding in a separate pull request that's fine too, otherwise I can take another look once you add that.